### PR TITLE
Add DCP0003 - Relative Lock Time Support.

### PR DESCRIPTION
**Pull request by davecgh**

### CODE REVIEW

File: dcp-0003/dcp-0003.mediawiki [hunk 1]
```diff
@@ -0,0 +1,837 @@
...
+(<code>CHECKSEQUENCEVERIFY</code>) for the Decred script system that provides
+the ability to conditionally enforce relative time-based restrictions, and
+changes the semantics for transaction time locks to use the median time of the
+previous 11 blocks instead of timestamp of the block itself in order to ensure
+the lock times increase monotonically with each block.
+
+==Motivation==
+
+There are a large number of use cases that either rely on, or can benefit from,
+the ability to prevent spending outputs until the have reached a certain
```
Comment (chappjc) [inline dcp-0003/dcp-0003.mediawiki]: the -> they

File: dcp-0003/dcp-0003.mediawiki [hunk 2]
```diff
@@ -0,0 +1,837 @@
...
+|-
+|invalid||2||4194304||empty||00b251||Mismatched lock types with min by-height argument and by-time sequence (argument == 0)
+|-
+|invalid||2||4194304||empty||03ffff00b251||Mismatched lock types with max by-height argument and by-time sequence (argument == 65535)
+|-
+|invalid||2||0||empty||03000040b251||Mismatched lock types with min by-time argument and by-height sequence (argument == 4194304)
+|-
+|invalid||2||0||empty||03ffff40b251||Mismatched lock types with max by-time argument and by-height sequence (argument == 4259839)
+|-
+|invalid||2||65535||empty||06000000000000b251||6-byte non-minally-encoded argument (argument == non-minimally-encoded 0)
```
Comment (chappjc) [inline dcp-0003/dcp-0003.mediawiki]: minally -> minimally

File: dcp-0003/dcp-0003.mediawiki [hunk 3]
```diff
@@ -0,0 +1,837 @@
...
+trustlessly closed at any time by either party by simply broadcasting the
+most recent commitment transaction to the network for inclusion in the
+blockchain.
+
+Since the commitment transactions are time locked, using absolute times would
+effectively place a lifetime on the channel.  This is the case because it
+would be necessary to close and reopen the channel any time the absolute timeout
+was approaching in order to prevent the counterparty from being able to steal
+funds.  This also implies that it would be necessary to use fairly long timeouts
+with absolute time locks in order to limit the frequency of channel churn.  Long
```
Comment (chappjc) [inline dcp-0003/dcp-0003.mediawiki]: Not clear to me what is meant by channel churn.  Creation and closing?

File: dcp-0003/dcp-0003.mediawiki [hunk 4]
```diff
@@ -0,0 +1,837 @@
...
+In short, off-chain atomic swaps provide the opportunity to create instant,
+trustless, decentralized cryptocurrency exchanges.
+
+===Escrow with Timeout===
+
+A common technique when making payments for products or services via an escrow
+is to create a 2-of-3 multsig transaction between the buyer, seller, and a
+third-party escrow agent.  In this type of setup, the escrow agent acts as an
+arbiter in the event the transaction does not go smoothly, but is not able to
+unilaterally steal the funds.
```
Comment (chappjc) [inline dcp-0003/dcp-0003.mediawiki]: Not sure if references here are appropriate, but perhaps to BitSquare.

**Review by chappjc — COMMENTED** [review]
.

File: dcp-0003/dcp-0003.mediawiki (hunk 3 above)
Comment (davecgh) [reply dcp-0003/dcp-0003.mediawiki]: Correct.  I changed it to say: "limit the frequency of opening and closing
channels".

File: dcp-0003/dcp-0003.mediawiki (hunk 4 above)
Comment (davecgh) [reply dcp-0003/dcp-0003.mediawiki]: I generally prefer not to link to commercial services from technical proposals (unless the proposal is actually based on the service or the service was specifically used for research and hence would be a reference).  Typically the references are limited to other technical documentation and whitepapers that were actually used in the creation of the proposal.  In this particular case, the escrow transactions are common knowledge that I didn't need to use any reference for, hence I didn't include anything.

**Comment by chappjc** [comment]
that's all for my review unless this hangs around all weekend.

File: dcp-0003/dcp-0003.mediawiki (hunk 1 above)
Comment (davecgh) [reply dcp-0003/dcp-0003.mediawiki]: Fixed.

File: dcp-0003/dcp-0003.mediawiki (hunk 2 above)
Comment (davecgh) [reply dcp-0003/dcp-0003.mediawiki]: Fixed.

**Pull request merged**